### PR TITLE
#66, #72: Refactor code for easier customization

### PR DESCRIPTION
**Pull request by martin-fleck-at**
- Use property injection over constructor injection
- Avoid readonly properties in classes that should be customizable

Fixes https://github.com/eclipse-glsp/glsp/issues/66
Fixes https://github.com/eclipse-glsp/glsp/issues/72

Signed-off-by: Martin Fleck <mfleck@eclipsesource.com>

### CODE REVIEW

File: packages/client/src/features/change-bounds/movement-restrictor.ts [hunk 1]
```diff
@@ -45,6 +45,8 @@ export function removeMovementRestrictionFeedback(element: SModelElement, moveme
 
 @injectable()
 export class NoOverlapMovmentRestrictor implements IMovementRestrictor {
+    public cssClasses = ["movement-not-allowed"];
```
Comment (tortmayr) [inline packages/client/src/features/change-bounds/movement-restrictor.ts:48]: [Very,very minor]: Why did you opt for the explicit `public` modifier her? AFAIK we omit `public` in most other cases and should probably keep it consistent.

File: packages/client/src/features/validation/validate.ts [hunk 2]
```diff
@@ -94,7 +93,7 @@ export function isSetMarkersAction(action: Action): action is SetMarkersAction {
 @injectable()
 export abstract class ExternalMarkerManager {
 
-    languageLabel: string;
+    public languageLabel: string;
```
Comment (tortmayr) [inline packages/client/src/features/validation/validate.ts:96]: Implicit vs. explicit public (consistency)

**Review by tortmayr — APPROVED** [review]
Thanks a lot. Code looks good to me. I also appreciate the additional effort to identify unused arguments and prefixing them with `_`.  :wink: 

I have raised one minor issue regarding implicit vs. explicit `public` but that does not prevent merging.
:+1:

File: packages/client/src/features/change-bounds/movement-restrictor.ts (hunk 1 above)
Comment (martin-fleck-at) [reply packages/client/src/features/change-bounds/movement-restrictor.ts:48]: Good point! I believe I added it explicitly as I had the feeling that we do add the `public` modifier more often than not, e.g., in all action constructors we use the public keyword. I am fine with both but it should be consistent. It is probably my Java background that tilts me a little bit more towards an explicit 'public' but I am also fine with omitting it. What do you think?

Comment (tortmayr) [reply packages/client/src/features/change-bounds/movement-restrictor.ts:48]: Constructors are a different story, constructor variables are private by default, here you need the explicit `public`.
I think we should opt for implicit public were applicable. We are heavily reusing code from Theia and sprotty and both use implicit public as well. So I guess, we should do the same.

Comment (martin-fleck-at) [reply packages/client/src/features/change-bounds/movement-restrictor.ts]: Good point! I rebased the branch and removed the explicit modifiers in a separate commit. Thank you for the explanation!

**Pull request merged**